test/e2e/csv_e2e_test.go

```diff
@@ -116,6 +116,12 @@ var csvReplacingChecker = func(csv *v1alpha1.ClusterServiceVersion) bool {
 	return csv.Status.Phase == v1alpha1.CSVPhaseReplacing || csv.Status.Phase == v1alpha1.CSVPhaseDeleting
 }
 
+var csvResourceVersionCheckerFor = func(csv *v1alpha1.ClusterServiceVersion) csvConditionChecker {
+	return func(fetchedCSV *v1alpha1.ClusterServiceVersion) bool {
+		return csv.GetResourceVersion() == fetchedCSV.GetResourceVersion()
```

Comment:
@ecordell what do you think of this approach to catching regressions like this?

Reply:
This will work if we're careful to use it correctly. Calling it twice in a row probably won't find the issues, but calling it once, waiting for a period of time that we don't expect changes to have occurred, and then calling it again will.

Reply:
I'm thinking we could use defer to our advantage here, like whenever we have an object hit a known state that we don't expect to change, go ahead and defer a check for the resourceVersion.

If that's not enough, we could keep a registry of resourceVersionChecks for the entire testsuite, and then check them as a final test.